src/Mvc/Mvc.Api.Analyzers/src/ActualApiResponseMetadataFactory.cs

```diff
@@ -84,9 +86,23 @@ void AnalyzeResponseExpression(IReturnOperation returnOperation)
         }
 
         var defaultStatusCodeAttribute = statementReturnType
-            .GetAttributes(symbolCache.DefaultStatusCodeAttribute, inherit: true)
+            .GetAttributes(defaultStatusCodeAttributeSymbol, inherit: true)
             .FirstOrDefault();
 
+        // If the type is not annotated with a default status code, then examine
+        // the attributes on any invoked method returning the type.
+        if (defaultStatusCodeAttribute is null && returnedValue.Syntax is InvocationExpressionSyntax targetInvocation)
+        {
+            var methodOperation = returnOperation.SemanticModel.GetSymbolInfo(targetInvocation);
+            var methodSymbol = methodOperation.Symbol ?? methodOperation.CandidateSymbols.FirstOrDefault();
+            if (methodSymbol is not null)
+            {
+                defaultStatusCodeAttribute = methodSymbol
+                    .GetAttributes()
```

Comment:
Does `methodSymbol.GetAttributes(defaultStatusCodeAttributeSymbol)` work? If not we could add the extension method. I would prefer it to avoid the delegate that `FirstOrDefault` currently allocates.

Reply:
`IMethodSymbol` doesn't have that particular overload for `GetAttributes` but we can add it.